src/codegen/generators/data_type_generator.rs

```diff
@@ -64,48 +66,49 @@ pub fn generate_data_types<'ink>(
                 .associate_type(name, llvm.create_struct_stub(struct_name).into())?;
         }
     }
+    // now create all other types (enum's, arrays, etc.)
     for (name, user_type) in types {
         let gen_type = generator.create_type(name, user_type)?;
         generator.types_index.associate_type(name, gen_type)?
     }
-    for (name, user_type) in types {
+
+    // extend the stubbed structs with the member fields
+    for (_, user_type) in types {
```

Comment:
could we get rid of the tuple here?

Reply:
done,
it can be joined with the following for-loop